### PR TITLE
Update minimum macOS support from 10.12 to 10.14

### PR DESCRIPTION
**Pull request by mike-myers-tob**
Closes https://github.com/osquery/osquery/issues/7574

1. changes the minimum deployment target on macOS as built by the CI workflow and in the top-level CMake file
2. documents the change in the user-facing build documentation (wiki pages), and I apologize but I slipped in a couple of fixes for Markdown nits unrelated to this change
3. documents the change in all of the developer notes for the third-party libraries
4. minor Markdown fixes and copy-editing in the changed notes from step 3
5. removed one special-case in the CMake for RocksDB that was required for macOS pre-10.14

### CODE REVIEW

**Review by directionless — APPROVED** [review]
I'm sad to skip High Sierra, but let's do it

File: libraries/cmake/source/augeas/README.md [hunk 1]
```diff
@@ -61,55 +62,49 @@ Copy the generated gnulib files:
  - `config.h` (use the augeas one)
  - All the files containing `AUTOGENERATED` under `lib`
 
-# macOS
```
Comment (Smjert) [inline libraries/cmake/source/augeas/README.md:64]: nit: Why two # here (and the additional # below)? This not a sub-heading of Linux

File: libraries/cmake/source/zstd/README.md [hunk 2]
```diff
@@ -29,18 +30,18 @@ cmake \
...
 Build the project
 
-```
+```bash
 cmake \
   --build build \
   -j $(nproc)
 ```
 
-# macOS
```
Comment (Smjert) [inline libraries/cmake/source/zstd/README.md:38]: nit: Same as above

File: libraries/cmake/source/augeas/README.md [hunk 3]
```diff
@@ -61,55 +62,49 @@ Copy the generated gnulib files:
...
+## macOS
```
Comment (Smjert) [inline libraries/cmake/source/augeas/README.md:65]: nit: Why two # here (and the additional # below)? This not a sub-heading of Linux

File: libraries/cmake/source/zstd/README.md [hunk 4]
```diff
@@ -29,18 +30,18 @@ cmake \
...
+## macOS
```
Comment (Smjert) [inline libraries/cmake/source/zstd/README.md:39]: nit: Same as above

File: libraries/cmake/source/augeas/README.md (hunk 3 above)
Comment (mike-myers-tob) [reply libraries/cmake/source/augeas/README.md:65]: In the Markdown files I edited that had more than one top-level heading I was trying to fix them but I may have made some mistakes, I'll go through again (thanks for spotting)

File: libraries/cmake/source/aws-sdk-cpp/README.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
-# Linux
+# aws-sdk-cpp library build notes
 
-## Common
+## Linux
+
+### Common
```
Comment (directionless) [inline libraries/cmake/source/aws-sdk-cpp/README.md:5]: What's common? And why is it under linux?

Comment (Smjert) [reply libraries/cmake/source/aws-sdk-cpp/README.md:5]: Normally the subheading under an OS refers to the arch, so common on both the arch we support

Comment (mike-myers-tob) [reply libraries/cmake/source/aws-sdk-cpp/README.md:5]: In this case, was my change incorrect?

Comment (directionless) [reply libraries/cmake/source/aws-sdk-cpp/README.md:5]: Sounds like it's fine -- it meets the structure stefano describes.

**Review by Smjert — APPROVED** [review]

**Pull request merged**